roachtest: sqlsmith ParseFloat value out of range

[(roachtest).sqlsmith/setup=seed/setting=no-mutations failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2509111&tab=buildLog) on [master@c41a2ed503cd31d95bc3ee4365663811772c8bd3](https://github.com/cockroachdb/cockroach/commits/c41a2ed503cd31d95bc3ee4365663811772c8bd3):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:181,sqlsmith.go:211,test_runner.go:760: error: pq: internal error: unexpected error from the vectorized engine: strconv.ParseFloat: parsing "-1.234E+401": value out of range
		stmt:
		WITH
			with_6285 (col_37139) AS (SELECT * FROM (VALUES (2409439561:::OID)) AS tab_15320 (col_37139)),
			with_6286 (col_37140)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								(
									st_box2dfromgeohash(e'aQ&\x1aP\f':::STRING::STRING, (-8263341481289712204):::INT8::INT8)::BOX2D
								),
								('BOX(-2.422136082476409 0.6784106693216416,-1.2086969764718063 1.9228445506173333)':::BOX2D),
								('BOX(-0.09765924723651909 -2.322449017006903,1.3712060537774633 1.0121650253902783)':::BOX2D)
						)
							AS tab_15321 (col_37140)
				)
		SELECT
			'88.215.184.229/0':::INET AS col_37141,
			((-1.234E+401):::DECIMAL::DECIMAL * '-53 years -10 mons -377 days -08:33:40.519057':::INTERVAL::INTERVAL)::INTERVAL
				AS col_37142,
			tab_15322._date AS col_37143,
			NULL AS col_37144,
			'6e22b025-5241-4b4a-8e43-9d3ecaf2343f':::UUID AS col_37145,
			'{"-MM{-BYq4@": [[]], "0IrMs s<w+m-": {":EEoiM": []}, "A5aAmJard": false, "a": 0.4766692198284659, "ix=3": {}}':::JSONB
				AS col_37146,
			tab_15322.crdb_internal_mvcc_timestamp AS col_37147,
			2826858748:::OID AS col_37148,
			tab_15322._int8 AS col_37149,
			tab_15322._float4 AS col_37150,
			tab_15322._int2 AS col_37151
		FROM
			defaultdb.public.seed@[0] AS tab_15322
		ORDER BY
			tab_15322._int4 ASC, tab_15322.tableoid
		LIMIT
			46:::INT8;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed/setting=no-mutations](https://teamcity.cockroachdb.com/viewLog.html?buildId=2509111&tab=artifacts#/sqlsmith/setup=seed/setting=no-mutations)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed%2Fsetting%3Dno-mutations.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/colexec/execgen/cmd/execgen/overloads_bin.go
+++ b/pkg/sql/colexec/execgen/cmd/execgen/overloads_bin.go
@@ -714,7 +714,7 @@ func (c intervalDecimalCustomizer) getBinOpAssignFunc() assignFunc {
 			return fmt.Sprintf(`
 		  f, err := %[3]s.Float64()
 		  if err != nil {
-		    colexecerror.InternalError(err)
+		    colexecerror.ExpectedError(err)
 		  }
 		  %[1]s = %[2]s.MulFloat(f)`,
 				targetElem, leftElem, rightElem)
@@ -732,7 +732,7 @@ func (c decimalIntervalCustomizer) getBinOpAssignFunc() assignFunc {
 			return fmt.Sprintf(`
 		  f, err := %[2]s.Float64()
 		  if err != nil {
-		    colexecerror.InternalError(err)
+		    colexecerror.ExpectedError(err)
 		  }
 		  %[1]s = %[3]s.MulFloat(f)`,
 				targetElem, leftElem, rightElem)

--- a/pkg/sql/colexec/proj_const_left_ops.eg.go
+++ b/pkg/sql/colexec/proj_const_left_ops.eg.go
@@ -11673,7 +11673,7 @@ func (p projMultDecimalConstIntervalOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := p.constArg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg.MulFloat(f)
 					}
@@ -11689,7 +11689,7 @@ func (p projMultDecimalConstIntervalOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := p.constArg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg.MulFloat(f)
 					}
@@ -11709,7 +11709,7 @@ func (p projMultDecimalConstIntervalOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := p.constArg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg.MulFloat(f)
 				}
@@ -11722,7 +11722,7 @@ func (p projMultDecimalConstIntervalOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := p.constArg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg.MulFloat(f)
 				}
@@ -14565,7 +14565,7 @@ func (p projMultIntervalConstDecimalOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := arg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = p.constArg.MulFloat(f)
 					}
@@ -14581,7 +14581,7 @@ func (p projMultIntervalConstDecimalOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := arg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = p.constArg.MulFloat(f)
 					}
@@ -14601,7 +14601,7 @@ func (p projMultIntervalConstDecimalOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := arg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = p.constArg.MulFloat(f)
 				}
@@ -14614,7 +14614,7 @@ func (p projMultIntervalConstDecimalOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := arg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = p.constArg.MulFloat(f)
 				}

--- a/pkg/sql/colexec/proj_const_right_ops.eg.go
+++ b/pkg/sql/colexec/proj_const_right_ops.eg.go
@@ -11674,7 +11674,7 @@ func (p projMultDecimalIntervalConstOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := arg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = p.constArg.MulFloat(f)
 					}
@@ -11690,7 +11690,7 @@ func (p projMultDecimalIntervalConstOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := arg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = p.constArg.MulFloat(f)
 					}
@@ -11710,7 +11710,7 @@ func (p projMultDecimalIntervalConstOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := arg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = p.constArg.MulFloat(f)
 				}
@@ -11723,7 +11723,7 @@ func (p projMultDecimalIntervalConstOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := arg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = p.constArg.MulFloat(f)
 				}
@@ -14566,7 +14566,7 @@ func (p projMultIntervalDecimalConstOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := p.constArg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg.MulFloat(f)
 					}
@@ -14582,7 +14582,7 @@ func (p projMultIntervalDecimalConstOp) Next(ctx context.Context) coldata.Batch 
 
 						f, err := p.constArg.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg.MulFloat(f)
 					}
@@ -14602,7 +14602,7 @@ func (p projMultIntervalDecimalConstOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := p.constArg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg.MulFloat(f)
 				}
@@ -14615,7 +14615,7 @@ func (p projMultIntervalDecimalConstOp) Next(ctx context.Context) coldata.Batch 
 
 					f, err := p.constArg.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg.MulFloat(f)
 				}

--- a/pkg/sql/colexec/proj_non_const_ops.eg.go
+++ b/pkg/sql/colexec/proj_non_const_ops.eg.go
@@ -12405,7 +12405,7 @@ func (p projMultDecimalIntervalOp) Next(ctx context.Context) coldata.Batch {
 
 						f, err := arg1.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg2.MulFloat(f)
 					}
@@ -12425,7 +12425,7 @@ func (p projMultDecimalIntervalOp) Next(ctx context.Context) coldata.Batch {
 
 						f, err := arg1.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg2.MulFloat(f)
 					}
@@ -12446,7 +12446,7 @@ func (p projMultDecimalIntervalOp) Next(ctx context.Context) coldata.Batch {
 
 					f, err := arg1.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg2.MulFloat(f)
 				}
@@ -12462,7 +12462,7 @@ func (p projMultDecimalIntervalOp) Next(ctx context.Context) coldata.Batch {
 
 					f, err := arg1.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg2.MulFloat(f)
 				}
@@ -15473,7 +15473,7 @@ func (p projMultIntervalDecimalOp) Next(ctx context.Context) coldata.Batch {
 
 						f, err := arg2.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg1.MulFloat(f)
 					}
@@ -15493,7 +15493,7 @@ func (p projMultIntervalDecimalOp) Next(ctx context.Context) coldata.Batch {
 
 						f, err := arg2.Float64()
 						if err != nil {
-							colexecerror.InternalError(err)
+							colexecerror.ExpectedError(err)
 						}
 						projCol[i] = arg1.MulFloat(f)
 					}
@@ -15514,7 +15514,7 @@ func (p projMultIntervalDecimalOp) Next(ctx context.Context) coldata.Batch {
 
 					f, err := arg2.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg1.MulFloat(f)
 				}
@@ -15530,7 +15530,7 @@ func (p projMultIntervalDecimalOp) Next(ctx context.Context) coldata.Batch {
 
 					f, err := arg2.Float64()
 					if err != nil {
-						colexecerror.InternalError(err)
+						colexecerror.ExpectedError(err)
 					}
 					projCol[i] = arg1.MulFloat(f)
 				}

--- a/pkg/sql/logictest/testdata/logic_test/vectorize_overloads
+++ b/pkg/sql/logictest/testdata/logic_test/vectorize_overloads
@@ -709,3 +709,7 @@ SELECT _int, _int2, _int // _int2 FROM many_types WHERE _int2 <> 0
 312  1  312
 789  4  197
 2    2  1
+
+# Regression test for incorrectly propagating an error as internal (#57773).
+statement error .* value out of range
+SELECT ((-1.234E+401)::DECIMAL * '-53 years -10 mons -377 days -08:33:40.519057'::INTERVAL::INTERVAL)::INTERVAL FROM many_types

